superset/models/sql_lab.py

```diff
@@ -179,6 +231,53 @@ def raise_for_access(self) -> None:
 
         security_manager.raise_for_access(query=self)
 
+    @property
+    def db_engine_spec(self) -> Type["BaseEngineSpec"]:
+        return self.database.db_engine_spec
+
+    @property
+    def owners_data(self) -> List[Dict[str, Any]]:
+        return []
+
+    @property
+    def uid(self) -> str:
+        return f"{self.id}__{self.type}"
+
+    @property
+    def is_rls_supported(self) -> bool:
+        return False
+
+    @property
+    def cache_timeout(self) -> int:
+        return 0
```

Comment:
the viz class checks if this is `None` and if not, it will look at the database. Should we set this to `None` instead?